components/service/glean/src/test/java/mozilla/components/service/glean/GleanTest.kt

```diff
@@ -224,14 +235,20 @@ class GleanTest {
             val expectedBaselineCounterMetrics = arrayOf(
                 "baseline.sessions"
             )
-            val baselineCounterMetrics = baselineJson.getJSONObject("metrics")!!.getJSONObject("counter")!!
+            val baselineCounterMetrics = baselineMetricsObject.getJSONObject("counter")!!
             assertEquals(expectedBaselineCounterMetrics.size, baselineCounterMetrics.length())
             for (metric in expectedBaselineCounterMetrics) {
                 assertNotNull(baselineCounterMetrics.get(metric))
             }
+
+            val baselineTimespanMetrics = baselineMetricsObject.getJSONObject("timespan")!!
+            assertEquals(1, baselineTimespanMetrics.length())
+            assertNotNull(baselineTimespanMetrics.get("duration"))
```

Comment:
Can we add a little manual delay, a 2-3 seconds maybe, between the ON_START and ON_STOP above and then validate that the recorded duration is valid?  This adds a little time to our test overhead but I think it's good to validate that it's recording a reasonable value.

Reply:
In general, we don't really want `sleep` in tests. I tried to fake the system clock using the `ShadowSystemClock`, but for some reason it's not updating the value returned by `elapsedRealtimeNanos`, which is used under the hood. I'd rather leave this as it is, since the timespan mechanism is being tested elsewhere.

Reply:
Okay, as long as it's being tested somewhere I'm good with that